src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1765,6 +1768,16 @@ static bool processOnChangeConditionForUpdateContext
       }
     }
   }
+  if (covered)
+  {
+    for (unsigned int ix = 0; ix < attrL.size(); ix++)
```

Comment:
As far as I understand, this loop iterates on all the elements in `notification.attributes` in the subscription, set each one to `null`.

However, what about on attributes that are also part of the entity? As far as I understand, they are added in the L1760-L1768 part, so this will cause "double adding".

To check this, I'd suggest to do a new .test with a slight modification in the subscriptions. Instead of:

```
payload='{
...
    "condition": {
      "attrs": [
        "pressure"
      ]
    }
  },
  "notification": {
    "http": {
      "url": "http://localhost:'$LISTENER_PORT'/notify"
    },
    "attrs": [
      "temperature",
      "humudity",
      "brightness"
    ],
...
}'
```

use:

```
payload='{
...
    "condition": {
      "attrs": [
        "pressure"
      ]
    }
  },
  "notification": {
    "http": {
      "url": "http://localhost:'$LISTENER_PORT'/notify"
    },
    "attrs": [
      "temperature",
      "pressure",
      "brightness"
    ],
...
}'
```

Reply:
Added new test in https://github.com/telefonicaid/fiware-orion/pull/4068/commits/a1770e3bc47943ca47b075ec25a9dd3ed512cda5

Reply:
The test in fine, but I'm still having doubts on this...

Anyway, I'd have a look with the debugger before merging this to master